### PR TITLE
fix: Solana handlers

### PR DESCRIPTION
**Pull request by Justkant**
### 📝 Description

- SignTransaction should not broadcast
- Allow any transaction that have a signer we can find (using the same logic as the device to find the signer in tx)
- Added signAllTransactions and signAndSendTransaction support
- Correctly deserialise the result from signTransaction
- Implement version-gated Solana support with semver validation

More details using AI:

Enable conditional Solana support based on Ledger Live version requirements using proper semantic versioning.

**Key Changes:**
- Add centralized `isSolanaSupportEnabled()` utility with semver-based version validation
- Refactor wallet request handlers to accept `walletInfo` parameter
- Add `walletInfo` parameter to `formatAccountsByChain` to enable version-based filtering
- Filter out Solana chains when wallet version doesn't meet minimum requirements
- Update `SessionProposal` and `SessionDetailEdit` components to use `walletInfosAtom`
- Update all `formatAccountsByChain` call sites in `useSupportedNamespaces` to pass `walletInfo`

**Dependencies:**
- Add `semver` and `@types/semver` for robust semantic version comparison

**Version Requirements:**
- Desktop: Ledger Live ≥ 2.126.0
- Mobile: Ledger Live ≥ 3.90.0

**Updated Components:**
- `src/hooks/requestHandlers/Wallet.ts`: Refactored `handleWalletRequest` and `getNetwork` functions
- `src/hooks/useWalletConnect.ts`: Updated to use `walletInfosAtom` and pass wallet info to handlers
- `src/hooks/useSupportedNamespaces.ts`: Simplified logic using direct atom access
- `src/utils/helper.util.ts`: Added version checking utilities with error handling

**Testing:**
- Comprehensive test coverage for version comparison edge cases
- Pre-release version support (beta, rc, alpha) following semver standards
- Invalid version format handling with graceful fallbacks
- Add wallet info mock to test suite and update test to include new parameter
- Ensure Solana chains are properly gated based on Ledger Live version constraints
- Add test for filtering Solana chains when desktop version is below minimum (2.126.0)
- Add test for filtering Solana chains when mobile version is below minimum (3.90.0)
- Add test for filtering Solana chains for unknown wallet types
- Verify that supported versions correctly include Solana chains in proposals
- Test mixed proposals with both Solana and non-Solana chains
- Ensure empty results when all Solana chains are filtered out

**Coverage includes:**
- Desktop wallet version boundaries (2.125.0 vs 2.127.0)
- Mobile wallet version boundaries (3.89.0 vs 3.91.0)
- Unknown wallet type handling
- Mixed chain proposals (Ethereum + Solana)
- Solana-only proposals

### ❓ Context

- **Linked resource(s)**: [LIVE-19307]

### 🚀 Expectations to reach

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-19307]: https://ledgerhq.atlassian.net/browse/LIVE-19307?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by qperrot — DISMISSED** [review]

File: src/hooks/requestHandlers/Solana.ts [hunk 1]
```diff
@@ -37,6 +42,28 @@ export async function handleSolanaRequest(
         walletKit,
       );
       break;
+    case SOLANA_SIGNING_METHODS.SOLANA_SIGN_ALL_TRANSACTIONS:
+      await signAllTransactions(
+        request,
+        topic,
+        id,
+        _chainId,
```
[the remark this answers is not in the document]
Comment (Justkant) [reply src/hooks/requestHandlers/Solana.ts:50]: I think it was done way before like that for other handlers that use it, we could remove it but I will need a new review just for this
